components/eventing-controller/controllers/backend/reconciler.go

```diff
@@ -800,6 +816,74 @@ func (r *Reconciler) deletePublisherProxy(ctx context.Context) error {
 	return err
 }
 
+func (r *Reconciler) updateMutatingValidatingWebhookWithCABundle(ctx context.Context) error {
+	// get the secret containing the certificate
+	var certificateSecret v1.Secret
+	secretKey := client.ObjectKey{
+		Namespace: certificateSecretNamespace,
+		Name:      certificateSecretName,
+	}
+	if err := r.Client.Get(ctx, secretKey, &certificateSecret); err != nil {
+		return utils.MakeError(errObjectNotFound, err)
+	}
+
+	// get the mutating and validation WH config
+	mutatingWH, validatingWH, err := r.getMutatingAndValidatingWebHookConfig(ctx)
+	if err != nil {
+		return err
+	}
+
+	// check that the mutating and validation WH config are valid
+	if len(mutatingWH.Webhooks) == 0 {
+		return utils.MakeError(errInvalidObject,
+			errors.Errorf("mutatingWH %s does not have associated webhooks", mutatingWHName))
+	}
+	if len(validatingWH.Webhooks) == 0 {
+		return utils.MakeError(errInvalidObject,
+			errors.Errorf("validatingWH %s does not have associated webhooks", validatingWHName))
+	}
+
+	// check if the CABundle present is valid
+	if mutatingWH.Webhooks[0].ClientConfig.CABundle != nil && validatingWH.Webhooks[0].ClientConfig.CABundle != nil {
+		if bytes.Equal(mutatingWH.Webhooks[0].ClientConfig.CABundle, certificateSecret.Data[tlsCertField]) &&
+			bytes.Equal(validatingWH.Webhooks[0].ClientConfig.CABundle, certificateSecret.Data[tlsCertField]) {
+			return nil
+		}
+	}
+
+	// update the ClientConfig for both the mutating and validation WH config
+	mutatingWH.Webhooks[0].ClientConfig.CABundle = certificateSecret.Data[tlsCertField]
+	err = r.Client.Update(ctx, mutatingWH)
```

Comment:
we could optimise it by having the condition in the line 848 only for mutating webhook. This means that if mutating webhook caBundle is up-to-date it doesn't need to update. 
Two mutating and validation webhook caBundle could be different if the update fails for any reason in the line 861.

Reply:
good point! will do it

Reply:
done